Core/src/main/java/com/paypal/android/core/api/models/APIResult.kt

```diff
@@ -0,0 +1,9 @@
+package com.paypal.android.core.api.models
+
+sealed class APIResult<out T> {
+    class Success<out T>(val data: T) : APIResult<T>()
+    class Failure(
+        val message: String? = null,
+        val throwable: Throwable? = null
+    ) : APIResult<Nothing>()
```

Comment:
should we keep this consistent with the rest of the SDK? Right now either we return an object or throw a `PayPalSDKError`. If the calling function is `suspend` we throw the `PayPalSDKError`. If its not we return it though a listener in `error()`